tests/OmniSharp.Roslyn.CSharp.Tests/FindImplementationFacts.cs

```diff
@@ -152,6 +152,38 @@ public class MyClass
             Assert.Equal(SymbolKind.Method, implementation.Kind);
         }
 
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task CanFindPartialClass(string filename)
+        {
+            const string code = @"
+                public partial class Some$$Class { void SomeMethod() {} }
+                public partial class SomeClass { void AnotherMethod() {} }";
+
+            var implementations = await FindImplementationsAsync(code, filename);
+
+            Assert.Equal(2, implementations.Count());
+            Assert.True(implementations.All(x => x.Name == "SomeClass"));
+        }
+
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task CanFindPartialMethod(string filename)
+        {
+            const string code = @"
+                public partial class SomeClass { partial void Some$$Method(); }
+                public partial class SomeClass { partial void SomeMethod() { /* this is implementation of the partial method */ }";
+
+            var implementations = await FindImplementationsAsync(code, filename);
+
+            Assert.Single(implementations);
+
+            var implementation = implementations.First();
+            Assert.Equal("SomeMethod", implementation.Name);
```

Comment:
Assert on the location of `implementation` to prove you found the implementation copy?

Reply:
this is a good idea.

Reply:
@SirIntruder if you could fix this feedback, this looks good to me.